lib/TPP/Conversion/ConvertLinalgToXsmm/ConvertLinalgToXsmm.cpp

```diff
@@ -808,6 +814,100 @@ void ConvertLinalgToXsmm::runOnOperation() {
     return signalPassFailure();
 }
 
+template <typename OpTy>
+static void updateGemmOpFlags(RewriterBase &rewriter, OpTy gemmDispatchOp) {
+  static_assert(llvm::is_one_of<OpTy, xsmm::GemmDispatchOp,
+                                xsmm::BrgemmDispatchOp>::value);
+  if (!gemmDispatchOp->hasOneUse())
+    return;
+  OpBuilder::InsertionGuard guard(rewriter);
+  rewriter.setInsertionPoint(gemmDispatchOp);
+
+  rewriter.updateRootInPlace(gemmDispatchOp, [&]() {
+    ArrayAttr flags = gemmDispatchOp.getFlags();
+    SmallVector<Attribute> newFlags;
+    for (auto flag : flags) {
+      if (auto gemmFlag = dyn_cast<xsmm::GemmFlagsAttr>(flag)) {
+        if ((gemmFlag.getValue() == xsmm::GemmFlags::NONE) ||
+            (gemmFlag.getValue() == xsmm::GemmFlags::BETA_0)) {
+          continue;
+        }
+      }
+      newFlags.push_back(flag);
+    }
+    newFlags.push_back(xsmm::GemmFlagsAttr::get(rewriter.getContext(),
+                                                xsmm::GemmFlags::BETA_0));
+    gemmDispatchOp.setFlagsAttr(rewriter.getArrayAttr(newFlags));
+  });
+}
+
+static void fuseZeroWithGemmOrBrgemm(RewriterBase &rewriter,
+                                     xsmm::UnaryOp rootOp) {
+  LLVM_DEBUG(llvm::dbgs() << "[fuseZeroWithGemmOrBrgemm] Candidate op: "
+                          << rootOp << "\n");
+  // 1. Prove the alloc to be zero. Linear walk of the bb
```

Comment:
There's no check that the same buffer is zeroed out. For example:
```
func.func @zero_test(%arg0: memref<1x32x32xf32>, %arg1: memref<1x32x32xf32>, %arg2: memref<32x32xf32>) {
  %c1_i64 = arith.constant 1 : i64
  %cst = arith.constant 0.000000e+00 : f32
  %alloc = memref.alloc() {alignment = 64 : i64} : memref<32x32xf32>
  %0 = xsmm.unary.dispatch zero [32, 512, 1, 512] flags = (bcast_scalar) data_type = f32
  xsmm.unary zero(data_type = f32, %0, %cst, %alloc) : (i64, f32, memref<32x32xf32>) -> ()
  %1 = xsmm.brgemm.dispatch [32, 32, 32, 32, 32, 32, 32, 1] flags = (none) data_type = f32
  // Sets incorrectly flags = (beta_0) when zero acts on %alloc not %arg2
  xsmm.brgemm(data_type = f32, %1, %arg0, %arg1, %arg2, %c1_i64) : (i64, memref<1x32x32xf32>, memref<1x32x32xf32>, memref<32x32xf32>, i64) -> ()
  return
}
```

Not saying we need to handle weird patterns but just avoid wrong codegen.